tests/lib/snaps/test-snapd-gpio-memory-control/gpiomem.c

```diff
@@ -0,0 +1,66 @@
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <termios.h>
+#include <unistd.h>
+
+#define MAP_SIZE 4096UL
+#define MAP_MASK (MAP_SIZE - 1)
+
+int main(int argc, char** argv)
+{
+    int fd;
+    void *map_base, *virt_addr;
+    uint32_t read_result, writeval;
+    off_t target;
+
+    if (argc < 2) {
+        printf("Usage: gpiomem ADDRESS [DATA]\n"
+               "\tADDRESS : memory address to act upon\n"
+               "\tDATA    : data to be written\n\n");
+        exit(1);
+    }
+    target = strtoul(argv[1], 0, 0);
```

Comment:
Nitpick, since the help message talks about `ADDRESS` this variable could be just called `address` too.